app/src/main/java/ch/epfl/qedit/view/answer/MatrixFragment.java

```diff
@@ -0,0 +1,107 @@
+package ch.epfl.qedit.view.answer;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.text.InputFilter;
+import android.text.InputType;
+import android.text.method.DigitsKeyListener;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.WindowManager;
+import android.widget.EditText;
+import android.widget.TableLayout;
+import android.widget.TableRow;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import ch.epfl.qedit.R;
+import ch.epfl.qedit.model.MatrixFormat;
+import java.util.ArrayList;
+
+public class MatrixFragment extends Fragment {
+    private TableLayout tableLayout;
+    private boolean hasDecimal = true;
+    private boolean hasSign = true;
+
+    private ArrayList<TableRow> tableRow = new ArrayList<>();
+    private ArrayList<ArrayList<EditText>> arrayButtons = new ArrayList<>();;
+
+    private int tableRowsNumber = 1;
+    private int tableColumnsNumber = 1;
+    private int maxCharacters = 3;
+    private String hintString;
+    private MatrixFormat matrixFormat;
+    private Context context;
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        matrixFormat = (MatrixFormat) getArguments().getSerializable("m0");
+        setValues();
+    }
+
+    // Used to get the context
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        this.context = context;
+    }
+
+    @Override
+    public void onDetach() {
+        super.onDetach();
+        context = null;
+    }
+
+    @Override
+    public View onCreateView(
+            @NonNull LayoutInflater inflater,
+            @Nullable ViewGroup container,
+            @Nullable Bundle savedInstanceState) {
+
+        View view = inflater.inflate(R.layout.answers_table, container, false);
+
+        tableLayout = view.findViewById(R.id.answersTable);
+
+        getActivity()
+                .getWindow()
+                .setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_HIDDEN);
+        for (int i = 0; i < tableRowsNumber; ++i) {
+            TableRow t = new TableRow(getActivity());
+            arrayButtons.add(new ArrayList<EditText>());
+            tableRow.add(t);
+            for (int j = 0; j < tableColumnsNumber; ++j) {
+                EditText editText = newEditText();
+                arrayButtons.get(i).add(editText);
+                tableRow.get(i).addView(editText);
+            }
+            tableLayout.addView(tableRow.get(i));
+        }
+
+        return view;
+    }
+
+    private void setValues() {
```

Comment:
We should remove this method I think, because it is useless to duplicate data around. It will just worsen the maintainability and readability of our code ;).

Reply:
I agree with the maintainability part but I don't agree with the readability part.